### ISSUE TITLE
sway freezes when exiting fullscreen 

### ISSUE THREAD

**Issue by GrownPlanet** [swaywm/sway#8104]
Sometimes, when exiting fullscreen in sway, my screen freezes and no inputs register, I need to physically shut down my pc and restart it to continue 

### Please fill out the following:
- **Sway Version:**
  - sway version 1.8.1

- **Debug Log:**
  - I wasn't able to get the log from sway (because the sway doesn't register inputs) itself bug here is the stack trace: [log](https://pastebin.com/Z31rK2DZ)

- **Configuration File:**
  - the bug also happens with the default configuration

- **Description:**
  - I don't know how to recreate the issue, but it has happened multiple times on multiple installs of fedora

**Comment by Lenny003** [comment]
I've been having this issue, thought it was my hardware freezing, but I had tested in xorg with i3 and never had it happen.

**Comment by Lenny003** [comment]
Yeah this has been happening to me too, I'm on Sway 1.9 and it still happens every now and again.

**Comment by kevenwyld** [comment]
I'm having basically exactly this same issue.  My system specs are:

sway version: `1:1.9-5`

```
OS: Arch Linux x86_64
Kernel: Linux 6.10.10-arch1-1
Shell: fish 3.7.1
Display (G274QPF-QD): 2560x1440 @ 165 Hz in 27″ [External]
Display (G274QPF-QD): 1440x2560 @ 165 Hz in 27″ [External]
Display (G274QPF-QD): 1440x2560 @ 165 Hz in 27″ [External]
Display (XR383CUR P): 3840x1600 @ 144 Hz in 38″ [External]
Display (Color LCD): 2048x1536 @ 60 Hz (as 1365x1024) in 10″ [External]
WM: Sway (Wayland)
Theme: Fusion [Qt], Adwaita-dark [GTK2/3]
Icons: Adwaita [Qt], Adwaita [GTK2/3]
Font: qt5ct [Qt], Nimbus Sans (10pt) [GTK2/3]
Cursor: Adwaita
Terminal: alacritty 0.13.2
Terminal Font: Source Code Pro (11pt)
CPU: AMD Ryzen 9 5950X (32) @ 3.40 GHz
GPU: AMD Radeon PRO W6800 [Discrete]
Memory: 8.19 GiB / 62.72 GiB (13%)
Swap: Disabled
Locale: en_US.UTF-8
```

I am unable to collect any logs because this appears to a complete and instant lockup.  No kernel events are recorded after the point which it freezes and I have to use the physical reset button.  I can not raise it over the network either.  The link is up but it won't respond to any connections.

I can recreate it on demand by using wl-gammarelay-rs (or any gamma color changing application, it also happens with all the other ones) to adjust the color temperature of all 5 screens simultaneously but this is not the only thing that causes it.  It also happens when physically connecting/disconnecting displays.  It even does this with gammarelay not installed.

AFAICT this must be made worse by the number of connected displays as I ran the system with a single monitor for a while and it was much less frequent.

I got it to be less bad by writing a script to change color temperature one display at a time but it will still freeze every other time if I try to change the 3840x1600 display directly.

It is also much less frequent on the linux-lts kernel (6.6.51-1) than on the current arch kernel (6.10.10.arch1-1).

I hope this information helps someone figure this out.  Thanks!
